discord/user.py

```diff
@@ -383,6 +393,21 @@ def blocked(self):
 
         http = self._state.http
 
+        try:
+            house = fields['house']
+            if house is None:
+                await http.leave_hypesquad_house()
+            elif 'first_name' not in fields or 'last_name' not in fields:
+                raise ClientException('When changing hypesquad houses first_name and last_name fields are required.')
```

Comment:
In my experience the first_name and last_name fields are neither required or sent by the official client.